staticcheck/lint.go

```diff
@@ -3817,3 +3819,58 @@ func CheckAddressIsNil(pass *analysis.Pass) (interface{}, error) {
 	code.Preorder(pass, fn, (*ast.BinaryExpr)(nil))
 	return nil, nil
 }
+
+var (
+	checkFixedLengthTypeShiftQ = pattern.MustParse(`
+		(Or
+			(AssignStmt _ (Or ">>=" "<<=") _)
+			(BinaryExpr _ (Or ">>" "<<") _))
+	`)
+	wordSize = gcsizes.ForArch(runtime.GOARCH).WordSize
```

Comment:
Sizeof returns bytes, not words. WordSize returns the size of a word in bytes (4 bytes or 8 bytes), not bits in a byte. That makes `pass.TypesSizes.Sizeof(typ) * wordSize` incorrect.

Reply:
Fixed.